vars/runATH.txt

```diff
@@ -0,0 +1,128 @@
+</ul>
+
+<pre>
+    runATH(metadataFile:"metadata.yml", athRevision: "master", athUrl:"https://github.com/jenkinsci/acceptance-test-harness.git", jenkins: "2.110")
+</pre>
+
+<p>
+    <b>Note that this step uses the <i>linux</i> and <i>docker && highmem</i> nodes in case that you need to run in different nodes
+    you can use the env variables <i>RUN_ATH_SOURCES_AND_VALIDATION_NODE</i> and <i>RUN_ATH_DOCKER_NODE</i> respectively to overwrite
+    the node to use</b>. Nodes are allocated automatically by the step if needed, but if the step is already running in the appropriate node no allocation is done
+</p>
+
+<p>
+    To make it usable in PR builders this step allows users to run the ATH using custom (typically previously built in the same Jenkinsfile) versions of any plugin, for that you need to set the metadata file's <i>useLocalSnapshots</i> property to true and stash the
+     plugins you want to use in the ATH run. By default you need to stash them with the name<i>localPlugins</i> the step will unstash them when appropriate and use the ATH`s <i>LOCAL_JARS</i> property to run the ATH. You can stash any number of plugins, all of
+     them will be used. You can also stash under other name by setting the env variable <i>RUN_ATH_LOCAL_PLUGINS_STASH_NAME</i>
+</p>
+
+<pre>
+    node("linux") {
+        dir("sources") {
+          checkout scm
+          List<String> mavenEnv = [
+                    "JAVA_HOME=${tool 'jdk8'}",
+                    'PATH+JAVA=${JAVA_HOME}/bin',
+                    "PATH+MAVEN=${tool 'mvn'}/bin"]
+          withEnv(mavenEnv) {
+            sh "mvn clean install -DskipTests"
+          }
+          dir("target") {
+           stash includes: '*.hpi', name: 'localPlugins'
+          }
+
+          runATH(metadataFile:"metadata.yml", athRevision: "master")
+        }
+    }
+</pre>
+
+<pre>
+    node("linux") {
+        dir("sources") {
+          checkout scm
+          List<String> mavenEnv = [
+                    "JAVA_HOME=${tool 'jdk8'}",
+                    'PATH+JAVA=${JAVA_HOME}/bin',
+                    "PATH+MAVEN=${tool 'mvn'}/bin"]
+          withEnv(mavenEnv) {
+            sh "mvn clean install -DskipTests"
+          }
+          dir("target") {
+           stash includes: '*.hpi', name: 'snapshots'
+          }
+
+          env.RUN_ATH_LOCAL_PLUGINS_STASH_NAME="snapshots"
+          runATH(metadataFile:"metadata.yml", athRevision: "master")
+        }
+    }
+</pre>
+<p>
+    The metadata file is a YAML file with the following structure:
+</p>
+
+<pre>
+---
+ath:
+  athUrl: https://github.com/jenkinsci/acceptance-test-harness.git
+  athRevision: acceptance-test-harness-1.59
+  jenkins: 2.89.4
+  failFast: false
+  rerunFailingTestsCount: 0
+  useLocalSnapshots: true
+  browsers:
+    - firefox
+    - chrome
+  tests:
+    - Test1
+    - Test2
+    - Test3
+  categories:
+    - Category1
+    - Category2
+</pre>
+
+<p>
+    Where:
+</p>
+
+<ul>
```

Comment:
This whole duplicated block looks like a maintenance burden? Can't we just link to this from the README and avoid duplication?

Reply:
As a user, I greatly appreciate when the "Pipeline syntax"  section offers me all the needed doc instead of a link to an external page, yeah I agree is a maintenance burden but I am thinking in users here

Reply:
We should aim at both. Or at least find some way for instance to convert one source to some other format in an automated fashion or so. Copy-pasting always ends up badly.